### DISCUSSION TITLE
is the provider baseUrl public ?

### DISCUSSION THREAD

**Discussion by Posoroko** [nuxt/image#1417] [Q&A]
When using a provider for images, the baseUrl must be set in the nuxt.config.ts file.  
Will this url be hidden from the public ? or will it be accessible ?

I'm proxying all request to my Directus backend and now I'm looking for a way to proxy the url used in the img tags, so it's hidden from the public

**Comment by yudin-s** [comment]
Yes, assume the provider `baseURL` is public.

Even if it is written in `nuxt.config.ts`, the generated image URL ultimately appears in the rendered HTML as an `img src` / `srcset`, so the browser has to know where to request it from.

If you do this:

```html
<NuxtImg src="/uploads/image.jpg" provider="directus" />
```

and the provider builds URLs against your Directus host, users can inspect the final URL in devtools.

If you want to hide the Directus origin, put a proxy in front of it and expose only your own URL, for example:

```txt
browser -> /_ipx/... or /images/... -> your server/proxy -> Directus
```

Then configure Nuxt Image/IPX/custom provider to generate URLs for your proxy route, not the Directus origin. That hides the upstream host from the browser, but the proxy endpoint itself is still public and should enforce whatever auth/rate limits you need.
